### PR TITLE
added IPv6 IPAM support

### PR DESCRIPTION
**Pull request by aanm**
ping @eyakubovich 

Signed-off-by: André Martins aanm90@gmail.com

### CODE REVIEW

File: plugins/ipam/host-local/main.go [hunk 1]
```diff
@@ -54,8 +56,17 @@ func cmdAdd(args *skel.CmdArgs) error {
...
-	r := &types.Result{
-		IP4: ipConf,
+	var r *types.Result
+
+	switch len(ipConf.IP.IP) {
+	case net.IPv6len:
+		r = &types.Result{
+			IP6: ipConf,
+		}
+	default:
```
Comment (steveeJ) [inline plugins/ipam/host-local/main.go]: would be nice to have a specific check for ipv4 and ipv6, and error out with an unknown length in the default case.

File: pkg/ipam/ipam.go [hunk 2]
```diff
@@ -39,6 +39,28 @@ func ExecDel(plugin string, netconf []byte) error {
...
+		return fmt.Errorf("failed to add addr to %q: %v", ifName, err)
+	}
+
+	for _, r := range ipConfig.Routes {
+		gw := r.GW
+		if gw == nil {
+			gw = ipConfig.Gateway
+		}
+		if err := ip.AddRoute(&r.Dst, gw, link); err != nil {
+			// we skip over duplicate routes as we assume the first one wins
```
Comment (steveeJ) [inline pkg/ipam/ipam.go]: in what cases do we have duplicate routes?

**Comment by steveeJ** [comment]
Thanks for this patch, and sorry for the review delay. I added a couple of comments.

File: pkg/ipam/ipam.go (hunk 2 above)
Comment (aanm) [inline pkg/ipam/ipam.go]: Not my code, that was already there. See [here](https://github.com/aanm/cni/blame/0becca2e96e3cf2fac1dc383e8890d617e306fe3/pkg/ipam/ipam.go#L66).
Ping @squaremo

File: plugins/ipam/host-local/main.go (hunk 1 above)
Comment (aanm) [inline plugins/ipam/host-local/main.go]: Makes sense, I'll take care of that.

**Comment by aanm** [comment]
Ping @steveeJ

**Comment by philips** [comment]
@aanm This comment doesn't look addressed: https://github.com/appc/cni/pull/110/files#r53413217

**Comment by philips** [comment]
@aanm This comment doesn't look addressed: https://github.com/appc/cni/pull/110/files#r53413217

**Comment by aanm** [comment]
@philips Somehow I missed that. Sorry. Fixed.

@steveeJ About the duplicate routes. That code was already there, I just move it to a function. I think that should be fixed in a different PR.

**Comment by steveeJ** [comment]
This will be superseded by #145 when it gets merged.

**Comment by bboreham** [comment]
> This will be superseded by #145 when it gets merged.

#145 was merged two months ago; shall we close this?

**Comment by rosenhouse** [comment]
Closing, see https://github.com/containernetworking/plugins/pull/12

**Pull request closed without merging**